website/CONTRIBUTE.md

```diff
@@ -0,0 +1,215 @@
+
+## Project structure
+
+```
+www/
+├── dist                                  # bundle files
+├── site
+│   ├── archetypes                        # frontmatter template
+│   ├── assets
+│   │   └── scss                          # styles
+│   ├── content                           # pages
+│   │   └── en
+│   │       ├── blog
+│   │       ├── community
+│   │       ├── contribute
+│   │       ├── documentation
+│   │       ├── get-started
+│   │       ├── privacy_policy
+│   │       ├── roadmap
+│   │       └── security
+│   │       └── _index.md
+│   ├── data
+│   ├── layouts                           # content template
+│   ├── static
+│   │   ├── downloads                     # downloaded files
+│   │   └── fonts
+│   │   └── images
+│   │   └── js
+│   └── themes
+│       └── docsy
+├── check-links.sh                        # links checker
+└── package.json
+```
+
+## How to add a new blogpost
+
+To add a new blogpost with pre-filled frontmatter, in `/www/site` run:
+
+```
+$ hugo new blog/my-new-blogpost.md
+```
+
+That will create a markdown file `/www/site/content/<LANGUAGE_VERSION>/blog/my-new-blogpost.md` with following content:
+
+```
+---
+title: "My New Blogpost"
+date: "2020-04-20T14:02:57+02:00"
+categories: 
+  - blog
+authors: 
+  - "Your Name"
+---
+```
+
+Below frontmatter, put your blogpost content. The filename will also serve as URL for your blogpost as `/blog/{filename}`.
+
+## How to add a new page
+
+For example, you would like to add a new `About` page.
+
+First, you need to create a markdown file in `/www/site/content/<LANGUAGE_VERSION>/about/_index.md` with following content:
+
+```
+---
+title: "Your page title"
+---
+```
+
+Below frontmatter, put your page content. The filename will also serve as URL for your page as `/about`.
+
+Second, define your page layout in the `layout` section with the same structure `/www/site/layout/about/{your_template}`. Hugo will help you to pick up the template behind the scene. Please refer to [Hugo documentation](https://gohugo.io/templates/) for the usage of templates.
+
+## How to write in Hugo
+
+This section will guide you how to use Hugo shortcodes in Apache Beam website. Please refer to [Hugo documentation](https://gohugo.io/content-management/shortcodes/) for more details of usage.
+
+### Define TableOfContents
+
+To automatically generate table of contents in a markdown file. Simply use:
+
+```
+{{< toc >}}
+```
+
+### Language switching
+
+To have a programming language tab switcher, for instance of java, python and go. Use:
+
+```
+{{< language-switchers java py go >}}
+```
+
+### Code highlighting
+
+To be consistent, please prefer to use `{{< highlight >}}` syntax instead of ` ``` `, for code-blocks or syntax-highlighters.
+
+1. To apply code highlighting to java, python or go. Use:
+
+```
+{{< highlight java >}}
+// This is java
+{{< /highlight >}}
+```
+
+2. To apply code highlighting to a wrapper class. Use:
+
+```
+{{< highlight class="runner-direct" >}}
```

Comment:
Maybe write something generic like `class="class-name"`, and include a resulting html below so that it's clear.